freqtrade/freqtradebot.py

```diff
@@ -549,7 +555,14 @@ def process_maybe_execute_sell(self, trade: Trade) -> bool:
 
             if trade.is_open and trade.open_order_id is None:
                 # Check if we can sell our current pair
-                return self.handle_trade(trade)
+                result = self.handle_trade(trade)
+
+                # Updating wallets if any trade occured
+                if result:
+                    self.wallets.update()
```

Comment:
calling this only here has a logical flaw ...
a) you'd want to call it also on/after sell
b) when `handle_trade` returns, a limit order may not be fulfilled yet, having balance return something other than what you expect it to. i would place this somewhere around the "`handle_timedout*`" functions ...

Reply:
Good point for timeout. but for handle_trade, I am not sure, as I get 'free': 

```
'BTC':   {           // string, three-letter currency code, uppercase
        'free': 321.00   // float, money available for trading
        'used': 234.00,  // float, money on hold, locked, frozen or pending
        'total': 555.00, // float, total balance (free + used)
    }

So 'free' should just consider capital on hold on a unfulfilled trade.